### ISSUE TITLE
Flatpak or AppImage Support?

### ISSUE THREAD

**Issue by ultimaplayer12** [Daivuk/apdoom#24]
Hello,

This is a request outside of the standard issues format because I am requesting support for a type of release. I'm currently using primarily my Steam Deck which runs a very restrictive OS based on Arch Linux. With this in mind, I'm unable to install third party libraries and some libraries are simply not included in the Steam Runtime. As such, APDoom is one program I've had difficulties getting running either through WINE/Proton, or the current Linux Pre-Release. It appears that pre-release is packaged primarily for Ubuntu. I would like to request that a maintainer packages it as either an AppImage with all required libraries included, or a Flatpak to be submitted to FlatHub. This could prevent issues from users on systems like mine, or systems where these libraries may not be built by default. I think it should be a priority to get support to as many users as possible, as DOOM has a rich community within the Open Source world and I very easily could see people using APDoom on a wide variety of *nix systems if a packaging system with included libraries was released.

Thank you!

**Comment by KScl** [comment]
While I've looked into AppImage bundles in the past, right now there are multiple things keeping the experience from being anything I'd call "good" on a Steam Deck, even if you _are_ willing to disable the read-only filesystem and install libfluidsynth and libsamplerate (which I did on my own Deck, for testing reasons; this is precisely why the current builds say that they are "not Steam Deck ready" yet).

- Starting games from the command line is reasonable on a PC, when you expect every user to have a keyboard ready for use. However, that's not a reasonable expectation for a Steam Deck user; ideally they should be able to start a game from game mode, and using the command line implicitly requires using desktop mode (and being quite familiar with it, I may add)
- Once you do get in game, it quickly becomes apparent that the controls are not well tested on controllers; it's workable, but it's definitely not ideal trying to navigate through the level select as it currently is
- Lastly, Doom and Heretic require separate binaries due to their extensive gameplay differences. This makes a launcher a _requirement_ instead of just a nice-to-have; there needs to be some central GUI to bring the user to when bundled, so that the user can, at the bare minimum, choose which executable they want to use

So basically, it's _planned_, but it'll be a little while before it's in a state where I feel comfortable with actually publishing an AppImage bundle.

**Comment by KScl** [comment]
Fixed by ArchipelagoDoom/APDoom@5d6870c. In-game menu controls still leave a bit to be desired, but changing that is a lower priority right now.
